### PR TITLE
Deferred register for MSEntityTypes

### PR DESCRIPTION
**Pull request by Dweblenod**
Changing our Entities to use a Deferred Register, also used the reformat code tool on entity classes

### CODE REVIEW

File: src/main/java/com/mraof/minestuck/entity/MSEntityTypes.java [hunk 1]
```diff
@@ -8,162 +8,106 @@
...
+	public static final RegistryObject<EntityType<ReturningProjectileEntity>> RETURNING_PROJECTILE = REGISTER.register("returning_projectile", () -> EntityType.Builder.<ReturningProjectileEntity>of(ReturningProjectileEntity::new, MobCategory.MISC).sized(0.25F, 0.25F).setTrackingRange(6).setUpdateInterval(2).build(new ResourceLocation(Minestuck.MOD_ID, "returning_projectile").toString())); //TODO smaller update interval value is temporary solution to improve client rendering
+	public static final RegistryObject<EntityType<BouncingProjectileEntity>> BOUNCING_PROJECTILE = REGISTER.register("bouncing_projectile", () -> EntityType.Builder.<BouncingProjectileEntity>of(BouncingProjectileEntity::new, MobCategory.MISC).sized(0.25F, 0.25F).setTrackingRange(6).setUpdateInterval(2).build(new ResourceLocation(Minestuck.MOD_ID, "bouncing_projectile").toString()));
 	
 	/**
 	 * Currently (1.15), this is not thread safe and need to be deferred
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/entity/MSEntityTypes.java:63]: Does the use of a deferred registry make it thread-safe? Can it be moved to public Minestuck(){}?

Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/entity/MSEntityTypes.java:63]: Nope, it has strictly to do with the implementation of `SpawnPlacements.register()`, so us switching to a deferred register does not matter here.

File: src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java [hunk 2]
```diff
@@ -2068,34 +2068,34 @@ protected void addTranslations()
...
-		addEntitySubtitles(MSEntityTypes.BASILISK, "ambient", "Basilisk hisses");
-		addEntitySubtitles(MSEntityTypes.BASILISK, "hurt", "Basilisk hurts");
-		addEntitySubtitles(MSEntityTypes.BASILISK, "death", "Basilisk dies");
-		addEntitySubtitles(MSEntityTypes.LICH, "ambient", "Lich wispers");
-		addEntitySubtitles(MSEntityTypes.LICH, "hurt", "Lich hurts");
-		addEntitySubtitles(MSEntityTypes.LICH, "death", "Lich dies");
-		addEntitySubtitles(MSEntityTypes.GICLOPS, "ambient", "Giclops wants to eat'cha");
-		addEntitySubtitles(MSEntityTypes.GICLOPS, "hurt", "Giclops hurts");
-		addEntitySubtitles(MSEntityTypes.GICLOPS, "death", "Giclops dies");
+		addEntitySubtitles(MSEntityTypes.NAKAGATOR.get(), "ambient", "Nakagator naks");
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java:2071]: Can you add an overload to `addEntitySubtitles()` that takes a supplier?

File: src/main/java/com/mraof/minestuck/entity/consort/EnumConsort.java [hunk 3]
```diff
@@ -13,10 +13,10 @@
 
 public enum EnumConsort	//TODO Could ideally be changed into a registry.
 {
-	SALAMANDER(() -> MSEntityTypes.SALAMANDER, "salamander", ChatFormatting.YELLOW, () -> MSSoundEvents.ENTITY_SALAMANDER_AMBIENT, () -> MSSoundEvents.ENTITY_SALAMANDER_HURT, () -> MSSoundEvents.ENTITY_SALAMANDER_DEATH),
-	TURTLE(() -> MSEntityTypes.TURTLE, "turtle", ChatFormatting.LIGHT_PURPLE, () -> null, () -> MSSoundEvents.ENTITY_TURTLE_HURT, () -> MSSoundEvents.ENTITY_TURTLE_DEATH),
-	NAKAGATOR(() -> MSEntityTypes.NAKAGATOR, "nakagator", ChatFormatting.RED, () -> MSSoundEvents.ENTITY_NAKAGATOR_AMBIENT, () -> MSSoundEvents.ENTITY_NAKAGATOR_HURT, () -> MSSoundEvents.ENTITY_NAKAGATOR_DEATH),
-	IGUANA(() -> MSEntityTypes.IGUANA, "iguana", ChatFormatting.AQUA, () -> MSSoundEvents.ENTITY_IGUANA_AMBIENT, () -> MSSoundEvents.ENTITY_IGUANA_HURT, () -> MSSoundEvents.ENTITY_IGUANA_DEATH);
+	SALAMANDER(MSEntityTypes.SALAMANDER::get, "salamander", ChatFormatting.YELLOW, () -> MSSoundEvents.ENTITY_SALAMANDER_AMBIENT, () -> MSSoundEvents.ENTITY_SALAMANDER_HURT, () -> MSSoundEvents.ENTITY_SALAMANDER_DEATH),
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/entity/consort/EnumConsort.java:16]: `RegistryObject<T>` implements `Supplier<T>`, so you should be able to use the registry object directly.

File: src/main/java/com/mraof/minestuck/entity/item/MetalBoatEntity.java [hunk 4]
```diff
@@ -146,8 +146,8 @@ public Packet<?> getAddEntityPacket()
 	
 	public enum Type
 	{
-		IRON(1/1.5F, () -> Items.IRON_INGOT, () -> MSItems.IRON_BOAT.get(), new ResourceLocation("minestuck", "textures/entity/iron_boat.png")),
-		GOLD(1.0F, () -> Items.GOLD_INGOT, () -> MSItems.GOLD_BOAT.get(), new ResourceLocation("minestuck", "textures/entity/gold_boat.png"));
+		IRON(1 / 1.5F, () -> Items.IRON_INGOT, MSItems.IRON_BOAT::get, new ResourceLocation("minestuck", "textures/entity/iron_boat.png")),
+		GOLD(1.0F, () -> Items.GOLD_INGOT, MSItems.GOLD_BOAT::get, new ResourceLocation("minestuck", "textures/entity/gold_boat.png"));
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/entity/item/MetalBoatEntity.java:150]: You should be able to use the registry object directly here.

File: src/main/java/com/mraof/minestuck/item/MSItems.java [hunk 5]
```diff
@@ -145,7 +145,7 @@
...
-	public static final RegistryObject<Item> TEARS_OF_THE_ENDERLICH_DRAWN = REGISTER.register("tears_of_the_enderlich_drawn", () -> new WeaponItem(new WeaponItem.Builder(MSItemTypes.CORUNDUM_TIER, 4, -1.5F).efficiency(10.0F).set(MSItemTypes.CLAWS_TOOL).set(ItemRightClickEffect.switchTo(MSItems.TEARS_OF_THE_ENDERLICH_SHEATHED)).add(OnHitEffect.targetSpecificAdditionalDamage(6, () -> MSEntityTypes.LICH)), new Item.Properties().defaultDurability(2000).tab(MSItemGroup.WEAPONS).rarity(Rarity.UNCOMMON)));
+	public static final RegistryObject<Item> TEARS_OF_THE_ENDERLICH_DRAWN = REGISTER.register("tears_of_the_enderlich_drawn", () -> new WeaponItem(new WeaponItem.Builder(MSItemTypes.CORUNDUM_TIER, 4, -1.5F).efficiency(10.0F).set(MSItemTypes.CLAWS_TOOL).set(ItemRightClickEffect.switchTo(MSItems.TEARS_OF_THE_ENDERLICH_SHEATHED)).add(OnHitEffect.targetSpecificAdditionalDamage(6, MSEntityTypes.LICH::get)), new Item.Properties().defaultDurability(2000).tab(MSItemGroup.WEAPONS).rarity(Rarity.UNCOMMON)));
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/item/MSItems.java:148]: You should be able to use the registry object directly here.

File: src/main/java/com/mraof/minestuck/world/lands/terrain/EndLandType.java [hunk 6]
```diff
@@ -38,7 +38,7 @@ public class EndLandType extends TerrainLandType
 	
 	public EndLandType()
 	{
-		super(new Builder(() -> MSEntityTypes.NAKAGATOR).names(END, DIMENSION)
+		super(new Builder(MSEntityTypes.NAKAGATOR::get).names(END, DIMENSION)
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/lands/terrain/EndLandType.java:41]: You should be able to use the registry object directly here and in other land types.

File: src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java (hunk 2 above)
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java]: Let me know if you think what I did is appropriate

File: src/main/java/com/mraof/minestuck/world/lands/terrain/RockLandType.java [hunk 7]
```diff
@@ -48,14 +48,14 @@ public class RockLandType extends TerrainLandType
...
-		return new RockLandType(Variant.ROCK, new Builder(() -> MSEntityTypes.NAKAGATOR).group(GROUP_NAME).names(ROCK, STONE, ORE)
+		return new RockLandType(Variant.ROCK, new Builder(MSEntityTypes.NAKAGATOR).group(GROUP_NAME).names(ROCK, STONE, ORE)
 				.skylight(7/8F).fogColor(0.5, 0.5, 0.55).skyColor(0.6, 0.6, 0.7)
 				.category(Biome.BiomeCategory.EXTREME_HILLS).music(() -> MSSoundEvents.MUSIC_ROCK));
 	}
 	
 	public static TerrainLandType createPetrification()
 	{
-		return new RockLandType(Variant.PETRIFICATION, new Builder(() -> MSEntityTypes.NAKAGATOR).group(GROUP_NAME).names(PETRIFICATION)
+		return new RockLandType(Variant.PETRIFICATION, new Builder(MSEntityTypes.NAKAGATOR::get).group(GROUP_NAME).names(PETRIFICATION)
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/lands/terrain/RockLandType.java:58]: Missed one here

Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/world/lands/terrain/RockLandType.java]: it is done

**Review by kirderf1 — APPROVED** [review]

**Pull request merged**